packages/utils/src/modules-sdk/abstract-service-factory.ts

```diff
@@ -237,6 +241,14 @@ export function abstractServiceFactory<
         sharedContext
       )
     }
+
+    @InjectTransactionManager(propertyRepositoryName)
+    async upsert(
+      data: (TDTOs["create"] | TDTOs["update"])[],
+      @MedusaContext() sharedContext: Context = {}
+    ): Promise<TEntity[]> {
+      return await this[propertyRepositoryName].restore(data, sharedContext)
```

Comment:
```suggestion
      return await this[propertyRepositoryName].upsert(data, sharedContext)
```

Reply:
ouch, nice catch 😂 don't know what happened to me at that moment